pkgs/tools/archivers/7zz/default.nix

```diff
@@ -1,47 +1,45 @@
-{ lib, stdenv, fetchurl, autoPatchelfHook }:
+{ stdenv, lib, fetchurl, gnugrep, p7zip }:
 
-let platform =       if stdenv.isi686    then "x86"
-                else if stdenv.isx86_64  then "x64"
-                else if stdenv.isAarch32 then "arm"
-                else if stdenv.isAarch64 then "arm64"
-                else throw "Unsupported architecture";
+# https://sourceforge.net/p/sevenzip/discussion/45797/thread/7fe6c21efa/
+stdenv.mkDerivation rec {
+  pname = "7zz";
+  version = "21.02";
 
-    url = "https://7-zip.org/a/7z2101-linux-${platform}.tar.xz";
+  src = fetchurl {
+    url = "https://7-zip.org/a/7z${lib.replaceStrings ["." ] [""] version}-src.7z";
+    sha256 = "sha256-7pdV/qoDTnB1cSvGsfrSVEC9Co8h8Qahbje8S0CfESI=";
+  };
 
-    hashes = {
-      x86 = "0k6vg85ld8i2pcv5sv3xbvf3swqh9qj8hf2jcpadssys3yyidqyj";
-      x64 = "1yfanx98fizj8d2s87yxgsy30zydx7h5w9wf4wy3blgsp0vkbjb3";
-      arm = "04iah9vijm86r8rbkhxig86fx3lpag4xi7i3vq7gfrlwkymclhm1";
-      arm64 = "0a26ginpb22aydcyvffxpbi7lxh4sgs9gb6cj96qqx7cnf7bk2ri";
-    };
-    sha256 = hashes."${platform}";
+  sourceRoot = "CPP/7zip/Bundles/Alone2";
 
-in stdenv.mkDerivation {
-  pname = "7zz";
-  version = "21.01";
+  # we need https://github.com/nidud/asmc/tree/master/source/asmc/linux in order
+  # to build with the optimized assembler but that doesn't support building with
+  # GCC
+  # "../../cmpl_gcc_x64.mak";
+  makefile = "../../cmpl_gcc.mak";
 
-  src = fetchurl { inherit url sha256; };
-  sourceRoot = ".";
+  NIX_CFLAGS_COMPILE = [ "-Wno-error=maybe-uninitialized" ];
 
-  nativeBuildInputs = [ autoPatchelfHook ];
-  buildInputs = [ stdenv.cc.cc.lib ];
+  nativeBuildInputs = [ gnugrep p7zip ];
 
-  dontBuild = true;
+  enableParallelBuilding = true;
 
   installPhase = ''
-    runHook preInstall
-    install -D -t $out/bin 7zz
-    runHook postInstall
+    install -Dm555 -t $out/bin b/g/7zz
+    install -Dm444 -t $out/share/doc/${pname} ../../../../DOC/*.txt
   '';
 
-  meta = with lib; {
-    description = "Command line archiver utility";
-    homepage = "https://www.7-zip.org";
+  doInstallCheck = true;
 
-    # source not released yet. will be under LGPL 2.1+ with RAR exception
-    license = licenses.unfree;
+  installCheckPhase = ''
+    $out/bin/7zz --help | grep ${version}
+  '';
 
-    platforms = [ "i686-linux" "x86_64-linux" "aarch64-linux" "armv7l-linux" ];
-    maintainers = with maintainers; [ anna328p ];
+  meta = with lib; {
+    description = "Command line archiver utility";
+    homepage = "https://7zip.org";
+    license = licenses.lgpl21Only;
+    maintainers = with maintainers; [ anna328p peterhoeg ];
+    platforms = platforms.linux;
```

Comment:
Not necessary for this PR, but it is possible to support macOS :thinking: ?

Reply:
I don't know - and I don't have hardware available on which to test.